karabo_bridge/client.py

```diff
@@ -87,32 +100,38 @@ def next(self):
 
     def _deserialize(self, msg):
         if len(msg) < 2:
-            return self._deserializer(msg[-1])
-
-        dat = {}
-        for header, data in zip(*[iter(msg)]*2):
+            data = self._deserializer(msg[-1])
+            meta = {}
+            for key, value in data.items():
+                meta[key] = value.get('metadata', {})
+            return data, meta
+
+        data = {}
+        meta = {}
+        for header, payload in zip(*[iter(msg)]*2):
             md = self._deserializer(header)
             source = md['source']
             content = md['content']
+            meta[source] = md.get('metadata', {})
```

Comment:
Do we replicate the same metadata in the header for each array? Or is it only in the first header with `content: "msgpack"` ?

Reply:
I just looked at the PR on PipeToZeroMQ, and it looks like the metadata is only on the first header, which is what I'd expect. So I think this is wrong, because it will replace it with an empty dict if there are any arrays!

Reply:
I see you point :)

Reply:
fixed